File: src/org/ggp/base/player/request/grammar/StopRequest.java
package org.ggp.base.player.request.grammar;

import java.util.List;

import org.ggp.base.player.gamer.Gamer;
import org.ggp.base.player.gamer.event.GamerCompletedMatchEvent;
import org.ggp.base.player.gamer.event.GamerUnrecognizedMatchEvent;
import org.ggp.base.player.gamer.exception.StoppingException;
import org.ggp.base.util.gdl.grammar.GdlTerm;
import org.ggp.base.util.logging.GamerLogger;

class StopRequest(Request):
{
    gamer = Gamer()
    matchId = ''
    private final List<GdlTerm> moves;

    def StopRequest(gamer=Gamer(), String matchId, List<GdlTerm> moves)
	{
        this.gamer = gamer;
        this.matchId = matchId;
        this.moves = moves;

    def getMatchId():  # String
        return matchId;

    def String process(int receptionTime)
	{
        // First, check to ensure that this stop request is for the match
        // we're currently playing. If we're not playing a match, or we're
        // playing a different match, send back "busy".
        if (gamer.getMatch() == null || !gamer.getMatch().getMatchId().equals(matchId))
		{
		    GamerLogger.logError("GamePlayer", "Got stop message not intended for current game: ignoring.");
            gamer.notifyObservers(new GamerUnrecognizedMatchEvent(matchId));
            return "busy";

		//TODO: Add goal values
        if(moves != null):
            gamer.getMatch().appendMoves(moves);
        gamer.getMatch().markCompleted(null);
        gamer.notifyObservers(new GamerCompletedMatchEvent());
        try {
            gamer.stop();
		} catch (StoppingException e):
		    GamerLogger.logStackTrace("GamePlayer", e);

		// Once the match has ended, set 'roleName' and 'match'
		// to NULL to indicate that we're ready to begin a new match.
        gamer.setRoleName(null);
	    gamer.setMatch(null);

        return "done";

    def String toString()
	{
        return "stop";
}

File: ggpy/cruft/autocode/StopRequest.py
#!/usr/bin/env python
""" generated source for module StopRequest """
# package: org.ggp.base.player.request.grammar
import java.util.List

import org.ggp.base.player.gamer.Gamer

import org.ggp.base.player.gamer.event.GamerCompletedMatchEvent

import org.ggp.base.player.gamer.event.GamerUnrecognizedMatchEvent

import org.ggp.base.player.gamer.exception.StoppingException

import org.ggp.base.util.gdl.grammar.GdlTerm

import org.ggp.base.util.logging.GamerLogger

class StopRequest(Request):
    """ generated source for class StopRequest """
    gamer = Gamer()
    matchId = str()
    moves = List()

    def __init__(self, gamer, matchId, moves):
        """ generated source for method __init__ """
        super(StopRequest, self).__init__()
        self.gamer = gamer
        self.matchId = matchId
        self.moves = moves

    def getMatchId(self):
        """ generated source for method getMatchId """
        return self.matchId

    def process(self, receptionTime):
        """ generated source for method process """
        #  First, check to ensure that this stop request is for the match
        #  we're currently playing. If we're not playing a match, or we're
        #  playing a different match, send back "busy".
        if self.gamer.getMatch() == None or not self.gamer.getMatch().getMatchId() == self.matchId:
            GamerLogger.logError("GamePlayer", "Got stop message not intended for current game: ignoring.")
            self.gamer.notifyObservers(GamerUnrecognizedMatchEvent(self.matchId))
            return "busy"
        if self.moves != None:
            self.gamer.getMatch().appendMoves(self.moves)
        self.gamer.getMatch().markCompleted(None)
        self.gamer.notifyObservers(GamerCompletedMatchEvent())
        try:
            self.gamer.stop()
        except StoppingException as e:
            GamerLogger.logStackTrace("GamePlayer", e)
        self.gamer.setRoleName(None)
        self.gamer.setMatch(None)
        return "done"

    def __str__(self):
        """ generated source for method toString """
        return "stop"

